### PR TITLE
COMPAS 2.x compatibility

### PR DESCRIPTION
**Pull request by gonzalocasas**
Update fab to COMPAS 2.x

### What type of change is this?

- [ ] Bug fix in a **backwards-compatible** manner.
- [ ] New feature in a **backwards-compatible** manner.
- [x] Breaking change: bug fix or new feature that involve incompatible API changes.
- [ ] Other (e.g. doc update, configuration, etc)

### Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I added a line to the `CHANGELOG.md` file in the `Unreleased` section under the most fitting heading (e.g. `Added`, `Changed`, `Removed`).
- [x] I ran all tests on my computer and it's all green (i.e. `invoke test`).
- [x] I ran lint on my computer and there are no errors (i.e. `invoke lint`).
- [x] I added new functions/classes and made them available on a second-level import, e.g. `compas_fab.robots.CollisionMesh`.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Review by yck011522 — APPROVED** [review]
lgtm

**Comment by yck011522** [comment]
@gonzalocasas Regarding the ur10e sample data. Do you want to use the new script to generate that package. The folder structure is not the same as the existing one.

**Review by chenkasirer — APPROVED** [review]
LGTM!
Left a few minor comments

File: src/compas_fab/data/planning_scene/cone_tool.json [hunk 1]
```diff
@@ -29,114 +29,4 @@
...
-                    "origin": null
-                }
-            ]
-        }
-    ],
-    "materials": [],
-    "name": "attached_tool"
-}
\ No newline at end of file
+                        "shape":
```
[the remark this answers is not in the document]
Comment (gonzalocasas) [reply src/compas_fab/data/planning_scene/cone_tool.json]: indeed! it was not used anymore, deleted at https://github.com/compas-dev/compas_fab/pull/403/commits/3e1c44771e8b01bb8b2aa5631feacd1d7318d75f

File: src/compas_fab/robots/trajectory.py [hunk 2]
```diff
@@ -245,15 +245,28 @@ def merged(self, other):
...
         self.planning_time = None
 
+    @property
+    def planning_time(self):
+        """:obj:`float` : Amount of time it took to complete the motion plan."""
+        return self.attributes["planning_time"]
+
+    @planning_time.setter
+    def planning_time(self, planning_time):
+        self.attributes["planning_time"] = planning_time
```
[the remark this answers is not in the document]
Comment (gonzalocasas) [reply src/compas_fab/robots/trajectory.py]: fixed at https://github.com/compas-dev/compas_fab/pull/403/commits/93d5837c1b1604c03943d47e60a6d5220db5b17a

File: src/compas_fab/robots/trajectory.py [hunk 3]
```diff
@@ -245,15 +245,28 @@ def merged(self, other):
 class Trajectory(Data):
```
[the remark this answers is not in the document]
Comment (gonzalocasas) [reply src/compas_fab/robots/trajectory.py:245]: also fixed at https://github.com/compas-dev/compas_fab/pull/403/commits/93d5837c1b1604c03943d47e60a6d5220db5b17a

File: src/compas_fab/backends/ros/fileserver_loader.py [hunk 4]
```diff
@@ -10,12 +10,11 @@
...
 from compas.datastructures import Mesh
-from compas.datastructures import mesh_transform
 from compas.files import XML
 from compas.geometry import Transformation
-from compas.robots.resources.basic import _get_file_format
-from compas.robots.resources.basic import _mesh_import
-from compas.utilities import geometric_key
+from compas.tolerance import TOL
+from compas_robots.resources.basic import _get_file_format
+from compas_robots.resources.basic import _mesh_import
```
[the remark this answers is not in the document]
Comment (gonzalocasas) [reply src/compas_fab/backends/ros/fileserver_loader.py:17]: true, I updated them here: https://github.com/compas-dev/compas_robots/pull/12

File: src/compas_fab/blender/scene.py
[the remark this answers is not in the document]
Comment (gonzalocasas) [reply src/compas_fab/blender/scene.py:1]: blender support is not complete, I'll fix that in a follow up PR

File: src/compas_fab/ghpython/reachabilitymapobject.py [hunk 5]
```diff
@@ -62,7 +62,7 @@ def draw_cloud(self, colormap="viridis", points=None):
 
         points = points or self.reachability_map.points
 
-        xpoints = [PrimitiveArtist(pt).draw() for pt in points]
+        xpoints = [SceneObject(pt).draw() for pt in points]
```
[the remark this answers is not in the document]
Comment (gonzalocasas) [reply src/compas_fab/ghpython/reachabilitymapobject.py]: good catch, fixed at https://github.com/compas-dev/compas_fab/pull/403/commits/a5d7a99f03d6d131aa33584eb376895a8e39c396

**Comment by gonzalocasas** [comment]
Except the blender parts, I think everything else has been addressed, so I will go ahead and merge this one! Thanks @chenkasirer and @yck011522 !!

**Pull request merged**